### PR TITLE
[JEP-229] Enable CD

### LINKED ISSUE #496: Use JEP-229
It is annoying to have to do reincrementalify `sample-plugin` after every release, and easily forgotten (https://github.com/jenkinsci/bom/commit/2b7c48813825243dc45b21f83319cfff0a591bcd was delayed by a week). See also https://issues.jenkins.io/browse/JENKINS-58641. Better to use JEP-229 to release on `workflow_dispatch` (I do not think we want automatic releases from this component).

### PR DESCRIPTION
**Pull request by jglick**
Fixes #496.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -147,18 +147,9 @@ so it is reasonable to retire BOMs for lines older than that.
...
-Due to a misconfiguration in Incrementals tooling (JENKINS-58641),
-currently after every release you must manually run
-
-```bash
-mvn -f sample-plugin incrementals:reincrementalify
-```
-
-Commit and push the result to fix the branch build.
+Automatic given [JEP-229](https://jenkins.io/jep/229) when PRs matching certain label patterns are merged.
+For the common case that only lots of `dependencies` PRs have been merged,
```
Comment (jglick) [inline README.md:151]: Given that #505 was an `enhancement`, I would expect merging _this_ PR to automatically trigger a release after a suitable delay (assuming no ci.jenkins.io outage of course).

File: README.md [hunk 2]
```diff
@@ -147,18 +147,9 @@ so it is reasonable to retire BOMs for lines older than that.
...
 
-`release:prepare` only runs basic tests about plugin versions, not the full PCT.
-Therefore be sure to check [commit status](https://github.com/jenkinsci/bom/commits/master)
-to ensure that CI builds have passed before cutting a release.
-
-Due to a misconfiguration in Incrementals tooling (JENKINS-58641),
-currently after every release you must manually run
-
-```bash
-mvn -f sample-plugin incrementals:reincrementalify
```
Comment (jglick) [inline README.md:158]: No more of an issue.

File: sample-plugin/pom.xml [hunk 3]
```diff
@@ -9,11 +9,10 @@
     </parent>
     <groupId>io.jenkins.tools.bom</groupId>
     <artifactId>sample</artifactId>
-    <version>${revision}${changelist}</version>
+    <version>${changelist}</version>
     <packaging>hpi</packaging>
     <properties>
-        <revision>30</revision>
-        <changelist>-SNAPSHOT</changelist>
+        <changelist>999999-SNAPSHOT</changelist>
```
Comment (jglick) [inline sample-plugin/pom.xml:15]: Still duplicated here, but overridden when built in release mode with `-Dset.changelist`, and should no longer ever need to be edited in sources.

File: sample-plugin/pom.xml [hunk 4]
```diff
@@ -343,17 +342,5 @@
                 <jenkins.version>2.222.4</jenkins.version>
             </properties>
         </profile>
-        <profile>
-            <id>skip-tests-on-release</id>
```
Comment (jglick) [inline sample-plugin/pom.xml:347]: No longer relevant due to `-Pquick-build` in https://github.com/jenkins-infra/jenkins-maven-cd-action/blob/b11e7369e037be500a81b3b02586b756c343ca4a/run.sh#L8.

File: .mvn/maven.config [hunk 5]
```diff
@@ -1,2 +1,3 @@
 -Pconsume-incrementals
 -Pmight-produce-incrementals
+-Dchangelist.format=%d.v%s
```
Comment (jglick) [inline .mvn/maven.config:3]: So now for example `805.vad4bd70bf54a`.

**Review by timja — APPROVED** [review]

**Review by jetersen — APPROVED** [review]

**Comment by jglick** [comment]
Seems to have worked! https://github.com/jenkinsci/bom/releases/tag/807.v6d348e44c987

**Pull request merged**